pymeasure/instruments/toptica/ibeamsmart.py

```diff
@@ -22,99 +22,267 @@
+            name,
             includeSCPI=False,
+            read_termination='\r\n',
+            write_termination='\r\n',
+            baud_rate=115200,
+            **kwargs
         )
+        # configure communication mode: no repeating and no command prompt
+        self.write('echo off')
+        self.write('prom off')
+        time.sleep(0.04)
+        # clear the initial messages from the controller
+        try:
+            self.adapter.flush_read_buffer()
+        except AttributeError:
+            log.warning("Adapter does not have 'flush_read_buffer' method.")
+        self.ask('talk usual')
+
+    def read(self):
+        """Read a reply of the instrument and exctract the values, if possible.
+
+        Reads a reply of the instrument which consists of at least two
+        lines. The initial ones are the reply to the command while the last one
+        should be '[OK]' which acknowledges that the device is ready to receive
+        more commands.
+
+        Note: '[OK]' is always returned as last message even in case of an
+        invalid command, where a message indicating the error is returned
+        before the '[OK]'
+
+        Value extraction: extract <value> from 'name = <value> [unit]'.
+        If <value> can not be identified the orignal string is returned.
+
+        :return: string containing the ASCII response of the instrument (without '[OK]').
+        """
+        reply = super().read()  # read back the LF+CR which is always sent back
+        if reply != "":
+            raise ValueError(
+                "TopticaAdapter.read(1): Error after command "
+                f"'{self.lastcommand}' with message '{reply}'")
+        msg = []
+        try:
+            while True:
+                line = super().read()
+                if line == '[OK]':
+                    break
+                msg.append(line)
+        except VisaIOError:
+            reply = '\n'.join(msg)
+            try:
+                self.adapter.connection.flush_read_buffer()
+            except AttributeError:
+                log.warning("Adapter does not have 'flush_read_buffer' method.")
+            raise ValueError(
+                "TopticaAdapter.read(2): Error after command "
+                f"'{self.lastcommand}' with message '{reply}'")
+        reply = '\n'.join(msg)
+        r = self._reg_value.search(reply)
+        if r:
+            return r.groups()[0]
+        else:
+            return reply
+
+    def write(self, command):
+        self.lastcommand = command
+        super().write(command)
+
+    def check_errors(self):
+        """Check communication after setting a value.
+
+        Checks if the last reply is only '[OK]', otherwise a ValueError is
+        raised and the read buffer is flushed because one has to assume that
+        some communication is out of sync.
+        """
+        reply = self.read()
+        if reply:
+            # anything else than '[OK]'.
+            self.adapter.connection.flush()
```

Comment:
The call signature of this function is wrong. 

See pyvisa documentation (https://pyvisa.readthedocs.io/en/latest/api/visalibrarybase.html#pyvisa.highlevel.VisaLibraryBase.flush). The mandatory argument is missing.

But in fact it would be better to replace to with a call to `self.adapter.flush_read_buffer()` (with the same error handling as above)

Reply:
Thanks for pointing it out. I changed it. Probably I mixed them up as I never used any of the both before.